datafusion/functions/src/lib.rs

```diff
@@ -84,7 +84,11 @@ use log::debug;
 #[macro_use]
 pub mod macros;
 
-make_package!(core, "core_expressions", "Core datafusion expressions");
+/// Core datafusion expressions
+/// Enabled via feature flag `core_expressions`
```

Comment:
This is the actual change. If this PR is merged, I will make follow on PR(s) to apply the same treatment to the other modules in `datafusion-functions`. 

I did only one initially to keep the size of the PR down